app/models/ability.rb

```diff
@@ -3,30 +3,52 @@ class Ability
 
   # Initializes the ability class
   def initialize(user)
-    # Order Abilities
-    # (Check https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence)
-    # Check roles of user, using rolify. Role name is *case sensitive*
-    # user.is_organizer? or user.has_role? :organizer
-    # user.is_cfp_of? Conference or user.has_role? :cfp, Conference
-    # user.is_info_desk_of? Conference
-    # user.is_volunteers_coordinator_of? Conference
-    # user.is_attendee_of? Conference
-    # The following is wrong because a user will only have 'cfp' role for a specific conference
-    # user.is_cfp? # This is always false
-
     user ||= User.new
 
-    # This is what sets up the different abilities
     if user.new_record?
       not_signed_in
-    # Checks if the user does not have any role and is not an admin
     elsif user.roles.any? || user.is_admin
-      signed_in_with_roles(user)
+      common_abilities_for_admins(user)
     else
       signed_in(user)
     end
```

Comment:
How about:
```ruby
if user.new_record?
  not_signed_in
else
  signed_in(user)
  common_abilities_for_admins(user) if user.roles.any? || user.is_admin?
end
```

And remove line 19: `signed_in(user)`.

This way, it is clearer to see that users with roles/admins also get the same permissions as signed in users.